backend/services/user_groups.go

```diff
@@ -0,0 +1,441 @@
+// Copyright 2022, Yahoo Inc.
+// Licensed under the terms of the MIT. See LICENSE file in project root for terms.
+
+package services
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/theparanoids/ashirt-server/backend"
+	"github.com/theparanoids/ashirt-server/backend/database"
+	"github.com/theparanoids/ashirt-server/backend/dtos"
+	"github.com/theparanoids/ashirt-server/backend/models"
+	"github.com/theparanoids/ashirt-server/backend/policy"
+	"github.com/theparanoids/ashirt-server/backend/server/middleware"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+type CreateUserGroupInput struct {
+	Name      string
+	Slug      string
+	UserSlugs []string
+}
+
+type ModifyUserGroupInput struct {
+	Name          string
+	Slug          string
+	UsersToAdd    []string
+	UsersToRemove []string
+}
+
+type ListUserGroupsForAdminInput struct {
+	UserGroupFilter
+	Pagination
+	IncludeDeleted bool
+}
+
+type ListUserGroupsForOperationInput struct {
+	Pagination
+	UserGroupFilter
+	OperationSlug string
+}
+
+type userGroupAndRole struct {
+	models.UserGroup
+	Role policy.OperationRole `db:"role"`
+}
+
+type ListUserGroupsInput struct {
+	Query          string
+	IncludeDeleted bool
+	OperationSlug  string
+}
+
+func (cugi ModifyUserGroupInput) validateUserGroupInput() error {
```

Comment:
the `cugi` should maybe be renamed to `mugi` (or maybe just `i` or something similar?).

Reply:
ooh good catch - yeah that was copy pasta